src/mod/ident.mod/ident.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * ident.c -- part of ident.mod
+ */
+/*
+ * MIT License
+ *
+ * Copyright (c) 2018 Michael Ortmann
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#define MODULE_NAME "ident"
+
+#include <errno.h>
+#include <fcntl.h>
+#include "src/mod/module.h"
+#include "server.mod/server.h"
+
+#define IDENT_METHOD_OIDENT  0
+#define IDENT_METHOD_BUILTIN 1
+
+static Function *global = NULL, *server_funcs = NULL;;
+
+static int ident_method = IDENT_METHOD_OIDENT;
+
+static tcl_ints identints[] = {
+  {"ident-method", &ident_method, 0},
+  {NULL,           NULL,          0}
+};
+
+static void ident_builtin_off();
+
+static cmd_t ident_raw[] = {
+  {"001", "",   (IntFunc) ident_builtin_off, "ident:001"},
+  {NULL,  NULL, NULL,                        NULL}
+};
+
+static void ident_builtin_off()
+{
+  rem_builtins(H_raw, ident_raw);
+}
+
+static void ident_activity(int idx, char *buf, int len)
+{
+  int s;
+  char buf2[128], *pos;
+  ssize_t i;
+
+  s = answer(dcc[idx].sock, &dcc[idx].sockname, 0, 0);
+  if ((i = read(s, buf2, 64)) < 0) {
+    putlog(LOG_MISC, "*", "Ident error: %s", strerror(errno));
+    return;
+  }
+  buf2[i - 1] = 0;
+  if (!(pos = strpbrk(buf2, "\n\r\\0"))) {
```

Comment:
Is the double backslash in `\\0` intentional?

Reply:
i think strpbrk as a string function cant take \0 as parameter anyway. its was coded too fast and ill need to rework some stuff, thats why i maked the PR as WIP now.

Reply:
You're right, `\0` wouldn't have the desired effect either. Just don't forget to change this, since it could cause some hard-to-diagnose issues with port numbers containing zeros.